### ISSUE TITLE
simulation code - step1, data distribution

### ISSUE THREAD

**Issue by martinjaggi** [epfml/DeAI#6]
provide a simulated decentralized code (not using any p2p backend, but instead just running locally), which holds a communication graph, and distributes a standard/toy ML dataset among the nodes. 

data distribution should support both random and heterogeneous / non-iid (for example different labels for each node). 

we can use standard PyTorch code examples e.g. MNIST or Cifar

**Comment by Saipraneet** [comment]
We can base this code off of this federated learning simulation codebase: https://github.com/AshwinRJ/Federated-Learning-PyTorch

The main training loop occurs here and it just uses nested foor loops: https://github.com/AshwinRJ/Federated-Learning-PyTorch/blob/master/src/federated_main.py

Reach out if you're interested in working on this: there are some gotchas with this codebase (bugs etc.)

**Comment by martinjaggi** [comment]
cool, this looks nice and compact. probably easy to run in colab too?

**Comment by Saipraneet** [comment]
Yep...I can also extract out the code from the personalization project with Mahmoud to have an easy starting point with all code in a colab.

**Comment by martinjaggi** [comment]
just pinging @ThomasDCepfl  here as he's working on it (based on the above excellent colab code by praneeth). he will keep us updated later...

**Comment by Saipraneet** [comment]
Committed the notebook: https://github.com/epfml/DeAI/commit/1e208c4c776d9df16c1f00467f078a78f305d7a4

Please feel free to re-open issue/unroll commit.
